HOTT-2306: Convert declarable formatted description text to links

maverickvi opened the pull request:
### Jira link

[HOTT-<2306>](https://transformuk.atlassian.net/browse/HOTT-2306)

### What?

I have added/removed/altered:

- [ ] Converted declarable formatted description text to links

### Why?

I am doing this because:

- it will allow users to easily navigate to headings, subheadings and chapters

<img width="1011" alt="Screenshot 2022-12-15 at 11 27 37" src="https://user-images.githubusercontent.com/12201130/207847909-95943566-95ad-44b2-85b1-03eb04edb86e.png">

maverickvi committed:
HOTT-2306: Convert declarable formatted description text to links

willfish commented on app/helpers/commodities_helper.rb:
```diff
@@ -58,6 +58,15 @@ def vat_overview_measure_duty_amounts(commodity)
   end
 
+  def convert_text_to_links(text)
+    base_url = request.base_url
```
I don't think you need the base_url?

willfish replied:
We can just use relative paths?

willfish commented on spec/helpers/commodities_helper_spec.rb:
```diff
@@ -182,4 +182,38 @@
+    context 'with 8 digit subheading in formatted description' do
+      let(:declarable_formatted_description) { ' 1234 11 22 ' }
+
+      it { is_expected.to eql " <a href='http://test.host/search?q=1234 11 22'>1234 11 22</a> " }
```
You'll want the query not to include spaces to be redirected to the subheading

willfish committed:
HOTT-2306: Make some changes to link converter:

1. Preserve query params from the previous request
2. Remove superfluous spaces that do not redirect properly
3. Remove base url and use relative paths
4. Update specs to reflect the changes

willfish committed:
HOTT-2306: Update beta search redirect and preserve country

willfish committed:
HOTT-2306: Lint

willfish committed:
HOTT-2306: Adds specs for the query method

willfish committed:
HOTT-2306: Fixes test failures

willfish committed:
Merge branch 'main' into hott-2306-comm-description-to-links
